integration/regression_tests.sh

```diff
@@ -0,0 +1,200 @@
+    exit 1
+fi
+
+# Apply default data choice and version if not set.
+echo "Using ${TEST_DATA_CHOICE:=${TEST_DATA_CHOICE_DEFAULT}} data."
+
+echo "Using version ${TEST_DATA_VERSION:=${TEST_DATA_VERSION_DEFAULT}} data."
+
+TEST_DATA_DIR=${TEST_DATA_PARENT_DIR}/${TEST_DATA_VERSION}/${TEST_DATA_CHOICE}
+
+if [ ! -d "${TEST_DATA_DIR}" ]; then
+    echo "ERROR: Test data directory \"${TEST_DATA_DIR}\" does not exist." >&2
+    exit 1
+fi
+
+echo "Binary equivalence/backwards compatibility test for um2nc."
+
+# Input paths
+source_ff=$TEST_DATA_DIR/fields_file
+
+# Reference netCDF files
+orig_nomask_nc=$TEST_DATA_DIR/reference_nomask.nc
+orig_mask_nc=$TEST_DATA_DIR/reference_mask.nc
+orig_hist_nc=$TEST_DATA_DIR/reference_hist.nc
+
+# Output paths
+out_nomask_nc=$OUTPUT_DIR/nomask.nc
+out_mask_nc=$OUTPUT_DIR/mask.nc
+out_hist_nc=$OUTPUT_DIR/hist.nc
+
+# -----------------------------------------------------------------
+# Functions and variables for running the tests
+# -----------------------------------------------------------------
+
+# Count the number of failed tests.
+N_TESTS_FAILED=0
+
+function clean_output {
+    echo "Removing test output files."
+    rm $out_mask_nc
+    rm $out_nomask_nc
+    rm $out_hist_nc
+}
+
+function run_um2nc {
+    # Run um2nc conversion. Exit if conversion fails.
+    ifile="${@: -2:1}"
+    echo "Converting \"${ifile}\"."
+    um2nc "$@"
+
+    if [ "$?" -ne 0 ]; then
+        echo "Conversion of \"${ifile}\" failed. Exiting." >&2
+        clean_output
+        exit 1
+    fi
+}
+
+function diff_warn {
+    # compare & warn if files do not match. Use nccmp flags passed in
+    # as arguments.
+    file1="${@: -2:1}"
+    file2="${@: -1:1}"
+    echo "Comparing \"$file1\" and \"$file2\"."
+    nccmp "$@"
+    if [ "$?" -ne 0 ]; then
+        (( N_TESTS_FAILED++ ))
+    else
+        echo "Files match."
+        echo
+    fi
+}
+
+# -----------------------------------------------------------------
+# Run the tests
+# -----------------------------------------------------------------
+
+# Test 1:
+# Execute nomask variant, pressure masking is turned OFF & all cubes are kept.
+run_um2nc    --nohist \
+             --nomask \
+             "$source_ff" \
+             "$out_nomask_nc"
+
+diff_warn -degh "$orig_nomask_nc"  "$out_nomask_nc"
+
+
+# Test 2:
+# Execute pressure masking variant: cubes which cannot be pressure masked are dropped.
+run_um2nc    --nohist \
+             "$source_ff" \
+             "$out_mask_nc"
+
+diff_warn -degh "$orig_mask_nc"  "$out_mask_nc"
+
+
+# Test 3:
+# Run without --nohist flag, and ignore history in nccmp comparison
+run_um2nc     \
+             "$source_ff" \
+             "$out_hist_nc"
+
+diff_warn -deg "$orig_hist_nc"  "$out_hist_nc"
+
+# Exit early if any comparisons failed.
+if [ $N_TESTS_FAILED -ne 0 ]; then
+    echo "${N_TESTS_FAILED} comparisons failed. netCDF output will be left at ${OUTPUT_DIR}." &>2
+    exit 1
+fi
```

Comment:
```suggestion
if [ -n "$FAILED_FILES" ]; then
    echo "Failed tests: ${#FAILED_FILES[@]}" &>2
    for files in ${FAILED_FILES[@]}; do
        echo "Failed comparison between \"${files/,/\" and \"}\"." # Using bash Parameter expansion with ${parameter/pattern/substitution}
    done
    exit 1
fi
```

Reply:
Good idea, much more informative to the user!